sympy/polys/galoistools.py

```diff
@@ -2328,15 +2328,122 @@ def _raise_mod_power(x, s, p, f):
     return linear_congruence(alpha, beta, p)
 
 
-def csolve_prime(f, p, e=1):
+def _csolve_prime_las_vegas(f, p, seed=1234):
```

Comment:
I recommend setting seed to None and only call `seed_func` if it is not None, that way the routine has freedom to give different answers, otherwise it is always locked into the same path unless the user manually selects a different path. Try follow the use in crypto as a pattern, e.g. `def elgamal_private_key(digit=10, seed=None):`

Reply:
np. Thanks for making the updates and learning a little more how we try handle random numbers. 

This is in.